File: go/property_config.go
// this file is @generated
package meteroid

// Type-specific configuration. Only the fields relevant to `property_type` are interpreted.
type PropertyConfig struct {
	Max *float64 `json:"max,omitempty"`

	// Maximum length for `TEXT`.
	MaxLength *int32 `json:"max_length,omitempty"`

	// Inclusive numeric bounds for `NUMBER`.
	Min *float64 `json:"min,omitempty"`

	// Allowed choices for `SINGLE_SELECT` / `MULTI_SELECT`.
	Options []SelectOption `json:"options,omitempty"`
}

File: python/meteroid/models/property_config.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .select_option import SelectOption


@dataclasses.dataclass
class PropertyConfig(BaseModel):
    """Type-specific configuration. Only the fields relevant to `property_type` are interpreted."""

    max: t.Optional[float] = None

    max_length: t.Optional[int] = None
    """Maximum length for `TEXT`."""

    min: t.Optional[float] = None
    """Inclusive numeric bounds for `NUMBER`."""

    options: t.Optional[t.List[SelectOption]] = None
    """Allowed choices for `SINGLE_SELECT` / `MULTI_SELECT`."""
